INTERFACE_INVENTORY.md

```diff
@@ -0,0 +1,266 @@
+```
+IReadOnlyUserRole : IReadOnlyUser
+```
+- `Role` (Role property)
+
+#### IReadOnlyProjectRole
+```
+IReadOnlyProjectRole : IReadOnlyProject
+```
+- `Role` (Role property)
+
+#### IReadOnlyPublisherRole
+```
+IReadOnlyPublisherRole : IReadOnlyPublisher
+```
+- `Role` (Role property)
+
+## Supporting Data Classes
+
+### Role
+```csharp
+public class Role
+{
+    public required string Id { get; init; }
+    public required string Name { get; init; }
+    public required string Description { get; init; }
+}
+```
+
+### Link
+```csharp
+public class Link : IStorable
+{
+    public required string Id { get; init; }
+    public required string Url { get; set; }
+    public required string Name { get; set; }
+    public required string Description { get; set; }
+}
+```
+
+### IReadOnlyConnection
+```csharp
+public interface IReadOnlyConnection
+{
+    string Id { get; }
+    Task<string> GetValueAsync(CancellationToken cancellationToken = default);
+    event EventHandler<string>? ValueUpdated;
+}
+```
+
+## Data Access Patterns for Blog Generator
+
+### For User Profiles
+1. **Basic Info**: `IReadOnlyEntity` → name, description, extendedDescription
+2. **Visual Elements**: `IReadOnlyImagesCollection` → profile images
+3. **External Links**: `IReadOnlyLinksCollection` → social/external profiles
+4. **App Connections**: `IReadOnlyConnectionsCollection` → connected applications
+5. **Project Involvement**: `IReadOnlyProjectRoleCollection` → projects and roles
+6. **Publisher Involvement**: `IReadOnlyPublisherRoleCollection` → publishers and roles
+
+### For Publisher Profiles
+1. **Basic Info**: `IReadOnlyEntity` → name, description, extendedDescription
+2. **Branding**: `IReadOnlyAccentColor` → brand color
+3. **Visual Elements**: `IReadOnlyImagesCollection` → logos, banners
+4. **External Links**: `IReadOnlyLinksCollection` → websites, social media
+5. **App Connections**: `IReadOnlyConnectionsCollection` → connected applications
+6. **Team Members**: `IReadOnlyUserRoleCollection` → users and their roles
+7. **Published Projects**: `IReadOnlyProjectCollection` → all projects under publisher
+8. **Publisher Hierarchy**: `ParentPublishers`, `ChildPublishers` → organizational structure
+
+### For Project Profiles
+1. **Basic Info**: `IReadOnlyEntity` → name, description, extendedDescription
+2. **Categorization**: `Category` → app store category
+3. **Branding**: `IReadOnlyAccentColor` → project theme color
+4. **Visual Elements**: `IReadOnlyImagesCollection` → screenshots, icons
+5. **External Links**: `IReadOnlyLinksCollection` → project website, documentation
+6. **App Connections**: `IReadOnlyConnectionsCollection` → integrations
+7. **Features**: `IReadOnlyFeaturesCollection` → feature list
+8. **Team Members**: `IReadOnlyUserRoleCollection` → contributors and roles
+9. **Dependencies**: `Dependencies` → other projects this depends on
+10. **Publisher**: `GetPublisherAsync()` → owning publisher
+
+## Interface Dependency Graph
```

Comment:
Malformed type dependency graph

Reply:
Interface graph is still insufficient. Create a graph for each of User, Project and Publisher separately, and include *all* missing interfaces inherited from the SDK.